mitodl/open-edx-plugins
# feat: add canvas integration plugin

### Related Tickets:
https://github.com/mitodl/open-edx-plugins/issues/4

### What's this PR does?
- Migrates the `canvas_integration` app from `edx-platform` into a separate plugin


### How to test?
To test this PR you will need a counterpart in the `edx-platform` and that part is specifically done in https://github.com/mitodl/edx-platform/pull/274

- Setup your edx-platform on the associated branch https://github.com/mitodl/edx-platform/pull/274 from here
- Install this plugin (e.g. generate a build for this plugin with `./pants package ::` and then install that into the edx-platform)
- Restart the lms (`make lms-restart`)
- Make sure to set the values for CANVAS_ACCESS_TOKEN=?, CANVAS_BASE_URL=https://mit.test.instructure.com` (You can ask me for the Token value)
(More instruction to be added once the PR is finalized)
- Now open a course as admin and click `Instructor` tab and make sure you see a `Canvas` tab under it
- Click the canvas tab and test all the functionality of canvas integration (e.g listing enrollments, syncing enrollments, pushing grades)

### Background context:
We had made several changes inside our for of the edx-platform where we added a Django application to support canvas integration. In the process, we made many changes in open edX's `Instructor` django application since we wanted to run `canvas_integration` inside the `Instrutor` tab. (More details to come)

### Screenshots:
**Landing page and tasks details**
![screencapture-localhost-18000-courses-course-v1-edX-DemoX-Demo-Course-instructor-2021-11-12-15_13_09](https://user-images.githubusercontent.com/34372316/141450297-05831367-eb09-4eb7-8943-12a0ef27439b.png)

**List Canvas Enrollments**
<img width="1394" alt="Screenshot 2021-11-12 at 3 14 15 PM" src="https://user-images.githubusercontent.com/34372316/141450488-01d179d1-ba89-4d18-8049-606d72d8a31e.png">

**List Canvas Assignments**
<img width="781" alt="Screenshot 2021-11-12 at 3 14 33 PM" src="https://user-images.githubusercontent.com/34372316/141450507-0f98f341-7403-4b75-8044-1ec007455a17.png">
